subsys/net/lib/coap/coap_client.c

```diff
@@ -820,8 +830,22 @@ static int handle_response(struct coap_client *client, const struct coap_packet
 void coap_client_cancel_requests(struct coap_client *client)
 {
 	for (int i = 0; i < ARRAY_SIZE(client->requests); i++) {
-		client->requests[i].request_ongoing = false;
+		if (client->requests[i].request_ongoing == true) {
+			LOG_DBG("Cancelling request %d", i);
+
+			coap_pending_clear(&client->requests[i].pending);
+
+			report_callback_error(&client->requests[i],
+					      COAP_RESPONSE_CODE_GATEWAY_TIMEOUT);
```

Comment:
Request force-cancelled seems different to me than a "regular" coap timeout. Isn't `-errno` codes already used for stuff like this? How about simply `-ECANCELED`?

Reply:
Done